reflex/components/radix/primitives/base.py

```diff
@@ -0,0 +1,23 @@
+"""The base component for Radix primitives."""
+from reflex.components.component import Component
+from reflex.components.tags.tag import Tag
+from reflex.utils import format
+from reflex.vars import Var
+
+
+class RadixPrimitiveComponent(Component):
+    """Basic component for radix Primitives."""
+
+    # Change the default rendered element for the one passed as a child.
+    as_child: Var[bool]
+
+    def _render(self) -> Tag:
+        return (
+            super()
+            ._render()
+            .add_props(
+                **{
+                    "class_name": format.to_title_case(self.tag or ""),
```

Comment:
i think the way this is written does not let the user pass their own `class_name` to the component.

probably we should have some code above like

```python
component_class = format.to_title_case(self.alias or self.tag or "")
class_name = (self.class_name or "") + component_class
```

and then call `add_props` with _that_ `class_name`

Reply:
I got this code from nihkil PR on the accordion, but I'm not even sure we need the component_class in the end.
Need to verify this.